internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -114,22 +119,48 @@ func (o *Orchestrator) Start() <-chan error {
 }
 
 // RunTask stops the current running task and starts task.
-func (o *Orchestrator) RunTask(task Task) {
+func (o *Orchestrator) RunTask(task Task, opts ...RunTaskOption) {
+	r := &runTaskAction{
+		task: task,
+	}
+	for _, opt := range opts {
+		opt(r)
+	}
+
 	// this guarantees the following:
-	// - if runTaskAction{} is pulled by the Orchestrator, any errors
+	// - if r is pulled by the Orchestrator, any errors
 	//   returned by it are reported by the Orchestrator.
 	// - if Stop() is called _before_ the Orchestrator picks up this
 	//   action, then this action is skipped.
 	select {
 	case <-o.stopped:
-	case o.actions <- &runTaskAction{
-		task: task,
-	}:
+	case o.actions <- r:
 	}
 }
 
 type runTaskAction struct {
-	task Task
+	task              Task
+	hosts             []Host
+	remoteContainerID string
+	network           net.IPNet
+}
```

Comment:
Given this configuration it seems non-obvious that these are optional. Maybe make a secondary struct:
```suggestion
type runTaskAction struct {
	task Task
	proxyOpts runTaskProxyOpts
}

type runTaskProxyOpts struct {
	hosts             []Host
	remoteContainerID string
	network           net.IPNet
}
```

Reply:
i added a comment!